src/main/java/org/owasp/webgoat/lessons/cryptography/HashingAssignment.java

```diff
@@ -51,7 +51,7 @@ public String getMd5(HttpServletRequest request) throws NoSuchAlgorithmException
 		String md5Hash = (String) request.getSession().getAttribute("md5Hash");
 		if (md5Hash == null) {
 			
-			String secret = SECRETS[new Random().nextInt(SECRETS.length)];
+			String secret = SECRETS[new java.security.SecureRandom().nextInt(SECRETS.length)];
 	         
 		    MessageDigest md = MessageDigest.getInstance("MD5");
```

Comment:
The MD5 algorithm is considered to be **completely** broken in security-related use cases. If you are using it to generate  tokens, hash passwords, or verify signatures, this algorithm should be changed. Because I'm just a bot, it's hard for me to tell if this should be changed -- but I wanted to bring this fact to your attention.

If you're looking for substitutes, consider `SHA-256` or `bcrypt`.

* https://en.wikipedia.org/wiki/Bcrypt